tests/projects/test_projects.py

```diff
@@ -72,21 +75,47 @@ def test_use_conda():
             os.environ["PATH"] = old_path
 
 
-def test_log_parameters():
-    """ Test that we log provided parameters when running a project. """
+def test_run():
+    for use_start_run in map(str, [True, False]):
+        with TempDir() as tmp, mock.patch("mlflow.tracking.get_tracking_uri")\
+                as get_tracking_uri_mock:
+            tmp_dir = tmp.path()
+            get_tracking_uri_mock.return_value = tmp_dir
+            run_uuid = mlflow.projects.run(
+                TEST_PROJECT_DIR, entry_point="test_tracking",
+                parameters={"use_start_run": use_start_run},
+                use_conda=False, experiment_id=0)
+            store = FileStore(tmp_dir)
+            run_infos = store.list_run_infos(experiment_id=0)
+            assert len(run_infos) == 1
+            store_run_uuid = run_infos[0].run_uuid
+            assert run_uuid == store_run_uuid
+            run = store.get_run(run_uuid)
+            expected_params = {"use_start_run": use_start_run}
+            assert run.info.status == RunStatus.FINISHED
+            assert len(run.data.params) == len(expected_params)
+            for param in run.data.params:
+                assert param.value == expected_params[param.key]
+            expected_metrics = {"some_key": 3}
+            for metric in run.data.metrics:
+                assert metric.value == expected_metrics[metric.key]
+
+
+def test_run_async():
     with TempDir() as tmp, mock.patch("mlflow.tracking.get_tracking_uri") as get_tracking_uri_mock:
         tmp_dir = tmp.path()
         get_tracking_uri_mock.return_value = tmp_dir
-        mlflow.projects.run(
-            TEST_PROJECT_DIR, entry_point="greeter", parameters={"name": "friend"},
-            use_conda=False, experiment_id=0)
-        store = FileStore(tmp_dir)
-        run_uuid = store.list_run_infos(experiment_id=0)[0].run_uuid
-        run = store.get_run(run_uuid)
-        expected_params = {"name": "friend"}
-        assert len(run.data.params) == len(expected_params)
-        for param in run.data.params:
-            assert param.value == expected_params[param.key]
+        run_uuid0 = mlflow.projects.run(
+            TEST_PROJECT_DIR, entry_point="sleep", parameters={"duration": 1},
+            use_conda=False, experiment_id=0, block=False)
+        assert tracking.get_run(run_uuid0).info.status == RunStatus.RUNNING
+        time.sleep(2)
+        assert tracking.get_run(run_uuid0).info.status == RunStatus.FINISHED
+        run_uuid1 = mlflow.projects.run(
+            TEST_PROJECT_DIR, entry_point="sleep", parameters={"duration": -1, "invalid-param": 30},
+            use_conda=False, experiment_id=0, block=False)
+        time.sleep(1)
```

Comment:
Are you sure this is enough time for it to crash? Might be good to give it more time here.